src/schema.rs

```diff
@@ -328,6 +346,18 @@ impl Serial for ModuleV1 {
     }
 }
 
+impl Serial for VersionedModuleSchema {
+    fn serial<W: Write>(&self, out: &mut W) -> Result<(), W::Err> {
+        match self {
+            VersionedModuleSchema::V1(module) => {
```

Comment:
So we have three schemas, named and ordered as:
```
schema v1
schema v2
versioned schema v1
```
Is that correct? 
We should add a comment about that, as the reuse of v1 is confusing without knowing the history.

Reply:
well, we have a schema for version 0 smart contracts, which can be found in custom section `concordium-schema-v1`.
We have two schemas for version 1 smart contracts, one in the section `concordium-schema-v2` without version information and one in `concordium-schema` with version information.

Technically schema type also have two versions for v1 smart contracts, since we have extended the type, but this is not tracked in the code

Reply:
I have added a bit more information on the types, but I am not sure how to convey all of this